core/lib/spree/core/engine.rb

```diff
@@ -15,6 +15,12 @@ class Engine < ::Rails::Engine
         generator.test_framework :rspec
       end
 
+      if ActiveRecord.respond_to?(:yaml_column_permitted_classes) || ActiveRecord::Base.respond_to?(:yaml_column_permitted_classes)
+        config.active_record.yaml_column_permitted_classes ||= []
+        config.active_record.yaml_column_permitted_classes
+          .concat([Symbol, BigDecimal, ActiveSupport::HashWithIndifferentAccess]).uniq!
```

Comment:
```suggestion
        config.active_record.yaml_column_permitted_classes |= [Symbol, BigDecimal, ActiveSupport::HashWithIndifferentAccess]
```

Looking at https://github.com/rails/rails/commit/8ce4bd1be83c08c30c34af4d0f1a726066128176, it looks like the setting already defaults to an empty array. Plus, we can leverage the [union operator](https://ruby-doc.org/core-3.1.0/Array.html#method-i-7C) to simplify the code. What do you think, @gsmendoza?

Reply:
@waiting-for-dev I I tested your suggestion and found that `config.active_record.yaml_column_permitted_classes` is nil by default. This is implied by [this line](https://github.com/rails/rails/commit/8ce4bd1be83c08c30c34af4d0f1a726066128176#diff-184a4c3f4a4726fe0cafd0f3504ca57f53a3d87e5315c5e34b58b830e9b5ef21R294) in the commit.

In contrast, [this line](https://github.com/rails/rails/commit/8ce4bd1be83c08c30c34af4d0f1a726066128176#diff-4411c7993e3f8ccfd8e09f44a81f10dfac3ea7dbd44cabd60b369040a00be324R164) is the one that sets `ActiveRecord::Base.yaml_column_permitted_classes` to an empty array.

I think the "config.active_record.yaml_column_permitted_classes is [] by default" spec in the commit should be corrected since it actually tests `ActiveRecord::Base.yaml_column_permitted_classes` instead of `config.active_record.yaml_column_permitted_classes` :/

As for the union operator, I don't think we can switch to it yet since it's only been introduced in Ruby 2.6. We still haven't removed Ruby 2.5 from Solidus (see https://github.com/solidusio/solidus/pull/4442).

Reply:
> @waiting-for-dev I I tested your suggestion and found that config.active_record.yaml_column_permitted_classes is nil by default.

I see. Thanks for checking.

> As for the union operator, I don't think we can switch to it yet since it's only been introduced in Ruby 2.6. We still haven't removed Ruby 2.5 support from Solidus (see https://github.com/solidusio/solidus/pull/4442).

I think it's also [available on 2.5](https://devdocs.io/ruby~2.5/array#method-i-7C) :slightly_smiling_face:

Reply:
> I think it's also [available on 2.5](https://devdocs.io/ruby~2.5/array#method-i-7C) slightly_smiling_face

@waiting-for-dev You're right :) It seems the `|` operator has been around for awhile. It's the `union` method that's only been added in [2.6.1](https://ruby-doc.org/core-2.6.1/Array.html). I'll update the configuration. Todo.